Add UserInputMatcher

atrzewik opened the pull request:
Grouped classes, reduced code notation

atrzewik committed:
Add UserInputMatcher

Grouped classes, reduced code notation

Czeffik commented on src/main/java/com/trzewik/blackjack/deck/Deck.java:
```diff
@@ -8,30 +8,30 @@
 public class Deck {
 
-    private ArrayList<Card> listOfCards = new ArrayList<>();
+    private ArrayList<Card> Cards = new ArrayList<>();
```
to nie jest nazwa klasy.... w Javie nazwy klas piszemy wielka litera, nazwy zmiennych małą

Czeffik commented on src/main/java/com/trzewik/blackjack/deck/Deck.java:
```diff
@@ -8,30 +8,30 @@
         shuffleDeck();
     }
 
     private ArrayList<Card> shuffleDeck(){
```
powinno byc voidem i tylko shufflowac, nie musisz niczego tutaj zwracac

Czeffik commented on src/main/java/com/trzewik/blackjack/game/Game.java:
```diff
@@ -58,9 +61,9 @@ private void playersBetting(){
     private MoveType getUserChoice(Player player){
         if (player.getCash() >= player.getBetValue() && player.getLastMove() == null){
-            return UserInputProvider.collectProperMoveType(new ArrayList<>(Arrays.asList(MoveType.STAND, MoveType.HIT, MoveType.DOUBLEDOWN)), new ArrayList<>(Arrays.asList(MoveType.STAND.getExpectedPlayerChoice(), MoveType.STAND.getExpectedPlayerChoice().toUpperCase(), MoveType.HIT.getExpectedPlayerChoice(), MoveType.HIT.getExpectedPlayerChoice().toUpperCase(), MoveType.DOUBLEDOWN.getExpectedPlay ...
+            return UserInputMatcher.collectProperMoveType(new ArrayList<>(Arrays.asList(MoveType.STAND, MoveType.HIT, MoveType.DOUBLE_DOWN)), MessageProvider.askPlayerForHitStandDouble, player.getName());
```
to co tutaj podajesz jako new ArrayList jest odpowiednikiem MoveType.values(), nie mozemy tego tutaj uzyc?

Czeffik commented on src/main/java/com/trzewik/blackjack/game/Game.java:
```diff
@@ -58,9 +61,9 @@ private void playersBetting(){
+            return UserInputMatcher.collectProperMoveType(new ArrayList<>(Arrays.asList(MoveType.STAND, MoveType.HIT, MoveType.DOUBLE_DOWN)), MessageProvider.askPlayerForHitStandDouble, player.getName());
         }
-        else {return UserInputProvider.collectProperMoveType(new ArrayList<>(Arrays.asList(MoveType.STAND, MoveType.HIT)), new ArrayList<>(Arrays.asList(MoveType.STAND.getExpectedPlayerChoice(), MoveType.STAND.getExpectedPlayerChoice().toUpperCase(), MoveType.HIT.getExpectedPlayerChoice(), MoveType.HIT.getExpectedPlayerChoice().toUpperCase())), MessageProvider.getAskPlayerForHitStand, player.getN ...
+        else {return UserInputMatcher.collectProperMoveType(new ArrayList<>(Arrays.asList(MoveType.STAND, MoveType.HIT)), MessageProvider.getAskPlayerForHitStand, player.getName());
```
chyba bedzie lepiej w MoveType'ie dorobic metode ktora bedzie zwracac MoveType[] ze standem i hitem

Czeffik commented on src/main/java/com/trzewik/blackjack/game/Game.java:
```diff
@@ -103,7 +106,7 @@ private void playersAuction(){
                     player.setLastMove(MoveType.STAND);
                 }
-                if (player.getLastMove() != MoveType.STAND && player.getLastMove() != MoveType.DOUBLEDOWN){
+                if (player.getLastMove() != MoveType.STAND && player.getLastMove() != MoveType.DOUBLE_DOWN){
```
jesli nie jest tym i nie jest tym, to jest czym? 😄  w tym wypadku chyba lepiej bedzie zrobic logike, ze jesli cos jest tym, to zrob to...

atrzewik replied:
a jeżeli lastMove== null? to wtedy też mam dwa warunki, że musi być tym lub tym

Czeffik replied:
tak, tylko wtedy to sie naturalniej czyta, nie uważasz?

Czeffik commented on src/main/java/com/trzewik/userinputprovider/UserInputMatcher.java:
```diff
@@ -0,0 +1,35 @@
+    public static MoveType collectProperMoveType(List<MoveType> moveTypes, String message, String ... formats){
+        while (true) {
+            try {
+                MoveType enumUserInput = collectMoveType(message,formats);
```
nie enumUserInput, tylko moveType 😃

Czeffik commented on src/main/java/com/trzewik/userinputprovider/UserInputMatcher.java:
```diff
@@ -0,0 +1,35 @@
+
+public class UserInputMatcher {
+
+    public static MoveType collectProperMoveType(List<MoveType> moveTypes, String message, String ... formats){
```
moveTypes -> properMoveTypes/expectedMoveTypes?

Czeffik commented on src/main/java/com/trzewik/userinputprovider/UserInputProvider.java:
```diff
@@ -32,31 +29,16 @@ public static Integer collectInteger(String message, String ... formats) {
     }
 
-    public static String collectProperString(List<String> listOfStrings, String message, String ... formats) {
+    public static String collectProperString(List<String> Strings, String message, String ... formats) {
```
to nie klasa, tylko zmienna (jw)

Czeffik commented on src/main/java/com/trzewik/userinputprovider/UserInputMatcher.java:
```diff
@@ -0,0 +1,35 @@
+            try {
+                String userInput = collectString(message, formats);
+                return MoveType.matchMove(userInput);
+            } catch (Exception e) {
```
IllegalArgumentExcpetion - tylko wtedy chcesz wyswietlic ten blad. Jak sie cos innego zjebie, to bedziemy to lapac inaczej 😄

Czeffik commented on src/main/java/com/trzewik/userinputprovider/UserInputMatcher.java:
```diff
@@ -0,0 +1,35 @@
+                    return  enumUserInput;
+                }
+                throw new IllegalArgumentException();
+            } catch (Exception e) {
```
IllegalArgumentException

Czeffik commented on src/main/java/com/trzewik/userinputprovider/UserInputProvider.java:
```diff
@@ -32,31 +29,16 @@ public static Integer collectInteger(String message, String ... formats) {
                     return userInput;
                 }
                 throw new IllegalArgumentException();
             } catch (Exception e) {
```
IllegalArgumentException

Czeffik reviewed (changes requested):
powoli sprzatamy ten balagan :D
nazwy branchy tez camel casem, albo chociaz tak jak w pythonie z `_`, tak zeby dalo sie to czytac.
czy MessageProvider na pewno musi byc w paczce game? predzej bym tam przeniosl MoveTypes, a dla providerow bym zrobil osobna paczke, np.: dataProviders

atrzewik committed:
Correct style

atrzewik committed:
Correct function

playerAuction corrected

atrzewik committed:
Correct function

UserInputMatcher collectProperMoveType corrected to an array

atrzewik committed:
Correct function

UserInputProvider collectProperString corrected style

atrzewik committed:
Correct Deck class

Deck corrected list creation

atrzewik committed:
Merge pull request #5 from atrzewik/userInputMatcherStyle

 Correct style

atrzewik committed:
Create contestants

Contestants created, contestants sorted by scores, contestants get positions and buster boolean, contestants sorted by positions, win prize created

atrzewik committed:
Create printResults

Contestants sorting by scores, setting busters and positions improved, printResults created, class MessageProvider and MessagePrinter developed

atrzewik committed:
Add enter after 'and'

atrzewik committed:
Improve printGameResults

setBusters changed to stream, deleted list of contestants positions, printGameResults corrected

atrzewik committed:
Change 'i'

atrzewik committed:
Test Sign, Card, Deck, Croupier

atrzewik committed:
Test Deck and Croupier improve

atrzewik committed:
Create setWinnersCash, checkIfBusters, stylistic corrections

atrzewik committed:
Overdue update

atrzewik committed:
Merge pull request #8 from atrzewik/tests

Test Sign, Card, Deck, Croupier

atrzewik committed:
Merge pull request #7 from atrzewik/createPrintResults

Create printResults

atrzewik committed:
Merge pull request #6 from atrzewik/contestantCreated

Create contestants
